pyquil/operator_estimation.py

```diff
@@ -930,38 +928,39 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
                 )
                 continue
 
-            # 3.3 Obtain statistics from result of experiment
-            obs_mean, obs_var = _stats_from_measurements(bitstrings, d_qub_idx, setting, n_shots, coeff)
+            # Obtain statistics from result of experiment
+            obs_mean, obs_var = _stats_from_measurements(bitstrings,
+                                                         {q: idx for idx, q in enumerate(qubits)},
+                                                         setting, n_shots, coeff)
 
             if calibrate_readout == 'plus-eig':
-                # 4 Readout calibration
-                # 4.1 Obtain calibration program
+                # Readout calibration
+                # Obtain calibration program
                 calibr_prog = _calibration_program(qc, tomo_experiment, setting)
-                # 4.2 Perform symmetrization on the calibration program
-                if symmetrize_readout == 'exhaustive':
-                    qubs_calibr = setting.out_operator.get_qubits()
-                    calibr_shots = n_shots
-                    calibr_results, d_calibr_qub_idx = _exhaustive_symmetrization(qc, qubs_calibr, calibr_shots, calibr_prog)
+                calibr_qubs = setting.out_operator.get_qubits()
+                calibr_qub_dict = {q: idx for idx, q in enumerate(calibr_qubs)}
 
-                else:
-                    raise ValueError("Readout symmetrization method must be either 'exhaustive' or None")
+                # Perform symmetrization on the calibration program
+                calibr_results = qc.run_symmetrized_readout(calibr_prog, n_shots, -1, calibr_qubs)
 
-                # 4.3 Obtain statistics from the measurement process
-                obs_calibr_mean, obs_calibr_var = _stats_from_measurements(calibr_results, d_calibr_qub_idx, setting, calibr_shots)
-                # 4.3 Calibrate the readout results
+                # Obtain statistics from the measurement process
+                obs_calibr_mean, obs_calibr_var = _stats_from_measurements(calibr_results,
+                                                                           calibr_qub_dict,
+                                                                           setting, n_shots)
+                # Calibrate the readout results
                 corrected_mean = obs_mean / obs_calibr_mean
                 corrected_var = ratio_variance(obs_mean, obs_var, obs_calibr_mean, obs_calibr_var)
 
                 yield ExperimentResult(
                     setting=setting,
                     expectation=corrected_mean.item(),
                     std_err=np.sqrt(corrected_var).item(),
-                    total_counts=n_shots,
+                    total_counts=len(bitstrings),
```

Comment:
i dont understand this code well enough to be sure, but this (and the similar two changes below) seems fishy to me -- can you explain the motivation for this change?

Reply:
The symmetrization code sometimes increases the number of shots above `n_shots`. If the symm_type is exhaustive and the number of qubits during calibration is different than for raw estimation (which I think should happen often), then the (potential) increase in shots could be different between `total_counts` and `calibration_counts`.

Reply:
But that was already the case before this PR right?

Reply:
Yes, you are right. I hadn't noticed that, but I believe this behavior is better? 

Also should I deprecate (or remove?) `_exhaustive_symmetrization` in this PR?

Reply:
If private functionality is completely superseded and no longer used, I see no issue with removing it!